### PR TITLE
gh-115315: Update time.rst to include microseconds field (%f) in chart

### PR DESCRIPTION
**Pull request by TahomaSoft**
Added documentation on microseconds and the %f field.

* Issue: gh-115315

----
📚 Documentation preview 📚: https://cpython-previews--115316.org.readthedocs.build/en/115316/library/time.html

### CODE REVIEW

File: Doc/library/time.rst
Comment (hugovk) [inline Doc/library/time.rst]: Build and lint fixes:

```suggestion
   | ``%f``    | Microseconds as a decimal number               |       |
```

**Review by hugovk — COMMENTED** [review]
See also https://github.com/python/cpython/issues/115315#issuecomment-1938596682.

File: Doc/library/time.rst
Comment (hugovk) [inline Doc/library/time.rst]: Remove trailing whitespace, link to functions, wrap to 80 cols:

```suggestion
       The ``%f` format directive only applies to :func:`strptime`,
       not to :func:`strftime`.
       However, see also :meth:`datetime.strptime` and
       :meth:`datetime.strftime`.
```

File: Doc/library/time.rst
Comment (hugovk) [inline Doc/library/time.rst]: Sorry, I missed a backtick, and we need the full ref for datetime:
```suggestion
   (3)
       The ``%f`` format directive only applies to :func:`strptime`,
       not to :func:`strftime`.
       However, see also :meth:`datetime.datetime.strptime` and
       :meth:`datetime.datetime.strftime`.
```

File: Doc/library/time.rst
Comment (hugovk) [inline Doc/library/time.rst]: This doesn't really tell us much. Why should we see those datetime methods?

By the way, please don't update the PR from `main` unless there's something important we need from there, or a conflict to resolve. It needlessly uses up extra CI time. Thanks!

Comment (TahomaSoft) [reply Doc/library/time.rst]: Ok, thanks for letting me know about the SOP (standard operating procedure) here about not updating the PR. Sorry about that. Other entities (some embedded Linux people) do want it updated.

How about:
However, see also :meth:`datetime.datetime.strptime` and
       :meth:`datetime.datetime.strftime` where the %f format directive applies to both methods in order to handle microseconds.

Alternatively, the whole "However ... " clause could be deleted.

Comment (hugovk) [reply Doc/library/time.rst]: > Ok, thanks for letting me know about the SOP (standard operating procedure) here about not updating the PR. Sorry about that. Other entities (some embedded Linux people) do want it updated.

No problem! Yeah, projects vary a lot :)

> How about: However, see also :meth:`datetime.datetime.strptime` and :meth:`datetime.datetime.strftime` where the %f format directive applies to both methods in order to handle microseconds.

Sounds good, how about:

```rst
However, see also :meth:`datetime.datetime.strptime` and
:meth:`datetime.datetime.strftime` where the ``%f`` format directive
:ref:`applies to microseconds <format-codes>`.
```

Comment (TahomaSoft) [reply Doc/library/time.rst]: Thanks!  Works for me. I'll make that change to my branch now.

Comment (TahomaSoft) [reply Doc/library/time.rst]: Sorry, didn't realize when I hit "resolve conversation" the CI checks would run again....

Comment (hugovk) [reply Doc/library/time.rst]: "Resolve" doesn't trigger the CI, pushing does -- and we want it to trigger for new pushes. Thanks!

**Review by hugovk — APPROVED** [review]
Thanks!

Let's give some time for the CODEOWNERS to review before merging.

**Comment by TahomaSoft** [comment]
Ok, thank you!Sent from my iPhoneOn Feb 13, 2024, at 14:44, Hugo van Kemenade ***@***.***> wrote:﻿
@hugovk approved this pull request.

Thanks!
Let's give some time for the CODEOWNERS to review before merging.

**Review by willingc — COMMENTED** [review]
Thanks @TahomaSoft. While we wait on feedback from the CODEOWNERS, it would be good to follow the alphabetical pattern of the table and move `%f` up in the table.

**Comment by TahomaSoft** [comment]
Will do!

> Thanks @TahomaSoft. While we wait on feedback from the CODEOWNERS, it would be good to follow the alphabetical pattern of the table and move `%f` up in the table.

**Comment by willingc** [comment]
Thanks @TahomaSoft for making the change and clarification.

@hugovk Do we still want to wait for CODEOWNERS on this doc PR or go ahead and merge? I think either would be fine.

**Review by willingc — APPROVED** [review]

**Comment by hugovk** [comment]
Let's merge. I think two weeks is long enough, we can easily make new changes later.

Thanks!

**Comment by TahomaSoft** [comment]
Thanks All!  Happy to have been able to contribute.

**Pull request merged**
